pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -222,9 +233,13 @@ pub mod pallet {
 			NextAssetId::<T>::set(next_asset_id);
 
 			// Should never fail.
-			let currency_ids =
-				BoundedVec::<FungiblesAssetIdOf<T>, T::MaxCurrencies>::try_from(currency_ids_vec.clone())
-					.map_err(|_| Error::<T>::CurrenciesNumber)?;
+			let currency_ids = BoundedVec::<FungiblesAssetIdOf<T>, T::MaxCurrencies>::try_from(
```

Comment:
Why not returning the bounded vec directly from `generate_sequential_asset_ids`?